orchestration/resource.go

```diff
@@ -0,0 +1,66 @@
+package orchestration
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"path"
+)
+
+type Resource struct {
+	client *Client
+}
+
+type ResourceGetRequest struct {
+	ResourceKey string `json:"resourceKey"`
+}
+
+type ResourceGetResponse struct {
+	ResourceName string `json:"resourceName"`
+	Version      int    `json:"version"`
+	VersionTag   string `json:"versionTag"`
+	ResourceId   string `json:"resourceId"`
+	TenantId     string `json:"tenantId"`
+	ResourceKey  string `json:"resourceKey"`
+}
+
+func (r Resource) Get(ctx context.Context, request ResourceGetRequest) (*ResourceGetResponse, error) {
+	u := r.client.baseURL
+	u.Path = path.Join(r.client.baseURL.Path, "resources", request.ResourceKey)
+	fmt.Println(u.Path)
+	requestBody, err := json.Marshal(request)
+	if err != nil {
+		return nil, fmt.Errorf("publish marshal: %w", err)
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), bytes.NewBuffer(requestBody))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := r.client.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		var getResponse ResourceGetResponse
+		err = json.NewDecoder(res.Body).Decode(&getResponse)
+		if err != nil {
+			return nil, fmt.Errorf("publish: %w", err)
+		}
+		return &getResponse, nil
+	default:
+		var errorResponse ErrorResponse
+		err = json.NewDecoder(res.Body).Decode(&errorResponse)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get error: %w", err)
```

Comment:
I would be specific here that it's about `decoding`. 

Something like this:
```suggestion
			return nil, fmt.Errorf("couldn't decode response: %w", err)
```